static/js/composebox_typeahead.js

```diff
@@ -841,6 +857,50 @@ exports.content_typeahead_selected = function (item, event) {
         // with the topic and the final **.
         const start = beginning.length - this.token.length;
         beginning = beginning.substring(0, start) + item + '** ';
+    } else if (this.completing === 'time_jump') {
+        const flatpickr_input = $("<input id='#timestamp_flatpickr'>");
+        let timeobject;
+        let timestring = beginning.substring(beginning.lastIndexOf('!time'));
+        if (timestring.startsWith('!time(') && timestring.endsWith(')')) {
+            timestring = timestring.substring(6, timestring.length - 1);
+            moment.suppressDeprecationWarnings = true;
+            try {
+                timeobject = moment(timestring).toDate();
+            } catch {
+                // do nothing. We'll just default to showing current date for the edit.
+            }
+        }
+
+        const instance = flatpickr_input.flatpickr({
+            enableTime: true,
+            clickOpens: false,
+            defaultDate: timeobject || moment().format(),
+            plugins: [new confirmDatePlugin({})], // eslint-disable-line new-cap, no-undef
+            positionElement: this.$element[0],
+            dateFormat: 'Z',
+            formatDate: (date) => {
+                const dt = moment(date);
+                return dt.local().format();
+            },
+        });
+        const container = $($(instance.innerContainer).parent());
+        container.on('click', '.flatpickr-calendar', (e) => {
+            e.stopPropagation();
+            e.preventDefault();
+        });
+
+        container.on('click', '.flatpickr-confirm', () => {
+            const datestr = flatpickr_input.val();
+            beginning = beginning.substring(0, beginning.lastIndexOf('!time')) +  `!time(${datestr}) `;
+            textbox.val(beginning + rest);
+            textbox.caret(beginning.length, beginning.length);
+            compose_ui.autosize_textarea();
+            instance.close();
+            instance.destroy();
+        });
+        instance.open();
+        container.find('.flatpickr-monthDropdown-months').focus();
+        return beginning + rest;
```

Comment:
The flatpickr calendar offers ranges, which we want to turn off.

It's also pretty ugly, but that's probably an issue for someone else :)

Reply:
Fixed in #15052 .